src/asl.ts

```diff
@@ -2482,6 +2407,15 @@ export class ASL {
       return undefined;
     } else if (
       mapOrParallelClosure === undefined ||
+      // TODO: this implementation is specific to the current state. Try/Catch/Finally needs to be generalize based on the generated ASL, not the AST.
+      // https://github.com/functionless/functionless/issues/385
+      (isArgument(mapOrParallelClosure.parent) &&
+        isCallExpr(mapOrParallelClosure.parent.parent) &&
+        isPropAccessExpr(mapOrParallelClosure.parent.parent.expr) &&
+        (mapOrParallelClosure.parent.parent.expr.name.name === "map" ||
+          mapOrParallelClosure.parent.parent.expr.name.name === "filter" ||
+          mapOrParallelClosure.parent.parent.expr.name.name === "forEach") &&
+        !isReferenceExpr(mapOrParallelClosure.parent.parent.expr.expr)) ||
```

Comment:
Is this fragile to ParnesExpr?

Reply:
wat?

Reply:
and yeah, this is fragile, (thus the TODO item). any other ideas?

Reply:
> wat?

I meant that it looks like it would fail if the argument was wrapped in parenthesized expression.